scripts/west_commands/runners/intel_adsp.py

```diff
@@ -0,0 +1,118 @@
+    '''Runner front-end for the intel ADSP boards.'''
+
+    def __init__(self,
+                 cfg,
+                 remote_host,
+                 timeout,
+                 ):
+        super().__init__(cfg)
+
+        self.remote_host = remote_host
+        self.timeout = timeout
+        self.bin_fw = os.path.join(cfg.build_dir, 'zephyr', 'zephyr.ri')
+
+        self.zephyr_base = os.environ.get('ZEPHYR_BASE')
+        self.cavstool = os.path.join(self.zephyr_base, DEFAULT_CAVSTOOL)
+        self.build_dir = cfg.build_dir
+        self.board_dir = cfg.board_dir
+        self.platform = os.path.basename(self.board_dir)
+
+    @classmethod
+    def name(cls):
+        return 'intel_adsp'
+
+    @classmethod
+    def capabilities(cls):
+        return RunnerCaps(commands={'flash'})
+
+    @classmethod
+    def do_add_parser(cls, parser):
+        # Optional:
+        parser.add_argument('--remote-host',
+                            help='hostname of the remote targeting ADSP board')
+        parser.add_argument('-t', '--timeout', type=int, default=60,
+                            help='the pty where stdout goes to')
+
+    @classmethod
+    def do_create(cls, cfg, args):
+        return IntelAdspBinaryRunner(cfg,
+                                    remote_host=args.remote_host,
+                                    timeout=args.timeout,
+                                    )
+
+    def do_run(self, command, **kwargs):
+        self.logger.info('Starting Intel ADSP runner')
+
+        if command == 'flash':
+            if re.search("intel_adsp_cavs", self.platform):
+                self.require(self.cavstool)
+                self.flash(**kwargs)
+            else:
+                self.run_sim(**kwargs)
+        else:
+            self.logger.info('Only support flash cmd now. Do nothing...')
+
+    def flash(self, **kwargs):
+        # Copy the zephyr and cavstool.py to target remote ADSP host
+        download_cmd = (['scp', f'{self.cavstool}', f'{self.bin_fw}', f'{self.remote_host}:'])
+        self.logger.info(f"cavstool({self.cavstool}), fw('{self.bin_fw})")
+
+        try:
+            self.check_call(download_cmd)
+        except subprocess.CalledProcessError:
+            return 1
+
+        # Excute the command "sudo ./cavstool.py zephyr.ri"
+        # FIXME: Make the cavstool.py works as a service and don't need
+        # to restart every time.
+        self.run_cmd = (['ssh', f'{self.remote_host}', 'sudo', './cavstool.py', 'zephyr.ri', '-t', f'{self.timeout}'])
+        self.kill_cmd = (['ssh', f'{self.remote_host}', 'sudo', 'pkill' , '-9', '-f', 'cavstool'])
+
+        self.logger.info(f"rcmd: {self.run_cmd}")
+
+        # We need this to cleanup the remote host process.
+        signal.signal(signal.SIGTERM, self.do_cleanup)
+
+        try:
+            self.check_call(self.run_cmd)
+        except Exception as e:
+            self.logger.error(e)
+            self.do_cleanup()
+            return 1
```

Comment:
Please don't swallow exceptions like this.

Reply:
Thanks for correcting this. Fix it.